yt/yt/core/logging/log_writer_detail.cpp

```diff
@@ -73,50 +73,128 @@ i64 TRateLimitCounter::GetAndResetLastSkippedEventsCount()
 
 ////////////////////////////////////////////////////////////////////////////////
 
-TStreamLogWriterBase::TStreamLogWriterBase(
-    std::unique_ptr<ILogFormatter> formatter,
-    TString name)
-    : Formatter_(std::move(formatter))
+TSegmentSizeReporter::TSegmentSizeReporter(TString logWriterName)
+    : CurrentSegmentSizeGauge_(
+        TProfiler{"/logging"}.WithSparse().WithTag("writer", std::move(logWriterName)).Gauge("/current_segment_size"))
```

Comment:
I am dangerously scared of using global variables in loggers, because a bunch of the logic happens during static initialization. I had my fun with siof while writing the code. Plus this is code by prime@, whom I tend to trust in these matters :) 

But I will give it some thought in hopes of convincing myself that using a global profiler would be fine.

Reply:
OK, no strong opinion here.

Reply:
I gave it some thought and decided that I do not want to interfere with the ability to configure `TLogManager` before main. Whether this is actually allowed now is debatable and depends on the current release (for example, loggers themselves became siof-safe in trunk). To be on the safer side while still introducing some cleanliness, I moved the profiler into a class field.